test/modules/TestResultCoordinator/Reports/EdgeHubRestartTest/EdgeHubRestartDirectMethodReportGenerator.cs

```diff
@@ -0,0 +1,303 @@
+            ulong senderResultCount,
+            ulong receiverResultCount)
+        {
+            List<TimeSpan> completedPeriods;
+            this.completedStatusHistogram.TryGetValue(HttpStatusCode.OK, out completedPeriods);
+            List<TimeSpan> orderedCompletedPeriods = completedPeriods?.OrderBy(p => p.Ticks).ToList();
+
+            TimeSpan minPeriod = TimeSpan.FromTicks(0);
+            TimeSpan maxPeriod = TimeSpan.FromTicks(0);
+            TimeSpan medianPeriod = TimeSpan.FromTicks(0);
+            TimeSpan meanPeriod = TimeSpan.FromTicks(0);
+            double variancePeriodInMilisec = 0.0;
+            if (orderedCompletedPeriods != null)
+            {
+                minPeriod = orderedCompletedPeriods.First();
+                maxPeriod = orderedCompletedPeriods.Last();
+
+                if ((orderedCompletedPeriods.Count & 0b1) == 0b1)
+                {
+                    // If odd, pick the middle value
+                    medianPeriod = orderedCompletedPeriods[orderedCompletedPeriods.Count >> 1];
+                }
+                else
+                {
+                    // If even, average the middle values
+                    medianPeriod =
+                        (orderedCompletedPeriods[orderedCompletedPeriods.Count >> 1] +
+                        orderedCompletedPeriods[(orderedCompletedPeriods.Count >> 1) - 1]) / 2;
+                }
+
+                // Compute Mean
+                TimeSpan totalSpan = TimeSpan.FromTicks(0);
+                double totalSpanSquareInMilisec = 0.0;
+                foreach (TimeSpan eachTimeSpan in orderedCompletedPeriods)
+                {
+                    totalSpan += eachTimeSpan;
+                    totalSpanSquareInMilisec += Math.Pow(eachTimeSpan.TotalMilliseconds, 2);
+                }
+
+                // Compute Mean : mean = sum(x) / N
+                meanPeriod = totalSpan / Math.Max(orderedCompletedPeriods.Count(), 1);
+
+                // Compute Sample Variance: var = sum((x - mean)^2) / (N - 1)
+                //                              = sum(x^2) / (N - 1) - mean^2
+                variancePeriodInMilisec = (totalSpanSquareInMilisec / Math.Max(orderedCompletedPeriods.Count() - 1, 1)) - Math.Pow(meanPeriod.TotalMilliseconds, 2);
+            }
+
+            return new EdgeHubRestartDirectMethodReport(
+                this.TrackingId,
+                this.TestReportType.ToString(),
+                passedDirectMethodCount,
+                this.SenderSource,
+                this.ReceiverSource,
+                senderResultCount,
+                receiverResultCount,
+                this.completedStatusHistogram,
+                minPeriod,
+                maxPeriod,
+                medianPeriod,
+                meanPeriod,
+                variancePeriodInMilisec);
+        }
+
+        async Task<(ulong resultCount, bool isNotEmpty, long sequenceNum)> IncrementSequenceNumberAsync(
```

Comment:
Please rename this since this method is not just increment sequence number at all.